blocks/autocompleters/style.scss

```diff
@@ -1,12 +1,35 @@
-.blocks-block-autocomplete .dashicon {
+.blocks-autocompleters__block .dashicon {
 	margin-right: 8px;
 }
 
-.blocks-user-autocomplete img {
-	margin-right: 8px;
-}
-
-.blocks-user-autocomplete .slug {
-	margin-left: 8px;
-	color: $dark-gray-100;
-}
+.blocks-autocompleters__user {
+	.blocks-autocompleters__user-avatar {
```

Comment:
One of the benefits of using a BEM convention is that it eliminates the need for nesting in many cases, so this could be moved to a top-level style definition.